scripts/build/setup/termux_setup_ghc.sh

```diff
@@ -14,26 +14,47 @@ termux_setup_ghc() {
 
 		export PATH="$TERMUX_GHC_RUNTIME_FOLDER/bin:$PATH"
 
-		if [ -d "$TERMUX_GHC_RUNTIME_FOLDER" ]; then return; fi
+		[ -d "$TERMUX_GHC_RUNTIME_FOLDER" ] && return
 
 		termux_download "https://downloads.haskell.org/~ghc/${TERMUX_GHC_VERSION}/ghc-${TERMUX_GHC_VERSION}-x86_64-deb10-linux.tar.xz" \
 			"$TERMUX_GHC_TAR" \
-			c1e31d798b013699b3c0de4fda27fb4cda47f572df0e75e3bd598a3012060615
+			a13719bca87a0d3ac0c7d4157a4e60887009a7f1a8dbe95c4759ec413e086d30
 
 		rm -Rf "$TERMUX_GHC_TEMP_FOLDER"
 		tar xf "$TERMUX_GHC_TAR" -C "$TERMUX_COMMON_CACHEDIR"
 
-		(set -e
+		(
+			set -e
 			unset CC CXX CFLAGS CXXFLAGS CPPFLAGS LDFLAGS AR AS CPP LD RANLIB READELF STRIP
 			cd "$TERMUX_GHC_TEMP_FOLDER"
 			./configure --prefix="$TERMUX_GHC_RUNTIME_FOLDER"
 			make install
 		)
 
-		rm -Rf "$TERMUX_GHC_TEMP_FOLDER"
+		# Cabal passes a host string to the libraries' configure scripts that isn't valid.
+		# After this patch we need to always pass --configure-option=--host=${TERMUX_HOST_PLATFORM}
+		# to Setup.hs configure.
+		(
+			CABAL_TEMP_FOLDER="$(mktemp -d cabal.XXXXXX)"
+			cd "${CABAL_TEMP_FOLDER}"
+
+			mkdir -p Cabal-3.6.2.0
+			curl -Ls \
+				https://hackage.haskell.org/package/Cabal-3.6.2.0/Cabal-3.6.2.0.tar.gz |
```

Comment:
Use the termux_download function to download it, since that function also verifies hashsum and checks so that it is not already downloaded